actor/manager/manager.go

```diff
@@ -0,0 +1,240 @@
+package manager
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/dapr/go-sdk/actor"
+	"github.com/dapr/go-sdk/actor/api"
+	"github.com/dapr/go-sdk/actor/codec"
+	actorErr "github.com/dapr/go-sdk/actor/error"
+	perrors "github.com/pkg/errors"
+	"reflect"
+	"sync"
+	"unicode"
+	"unicode/utf8"
+)
+
+type ActorManager interface {
+	RegisterActorImplFactory(f actor.Factory)
+	InvokeMethod(actorID, methodName string, request []byte) ([]byte, actorErr.ActorErr)
+	DetectiveActor(actorID string) actorErr.ActorErr
+	InvokeReminder(actorID, reminderName string, params []byte) actorErr.ActorErr
+	InvokeTimer(actorID, timerName string, params []byte) actorErr.ActorErr
+}
+
+// DefaultActorManager is to manage one type of actor
+type DefaultActorManager struct {
+	// factory is the actor factory of specific type of actor
+	factory actor.Factory
+
+	// activeActors stores the map actorID -> ActorContainer
+	activeActors sync.Map
+
+	// serializer is the param and response serializer of the actor
+	serializer codec.Codec
+}
+
+func NewDefaultActorManager(serializerType string) (ActorManager, actorErr.ActorErr) {
+	serializer, err := codec.GetActorCodec(serializerType)
+	if err != nil {
+		return nil, actorErr.ErrActorSerializeNoFound
+	}
+	return &DefaultActorManager{
+		serializer: serializer,
+	}, actorErr.Success
+}
+
+// RegisterActorImplFactory registers the action factory f
+func (m *DefaultActorManager) RegisterActorImplFactory(f actor.Factory) {
+	m.factory = f
+}
+
+// getAndCreateActorContainerIfNotExist will
+func (m *DefaultActorManager) getAndCreateActorContainerIfNotExist(actorID string) ActorContainer {
+	val, ok := m.activeActors.Load(actorID)
+	if !ok {
+		m.activeActors.Store(actorID, NewDefaultActorContainer(actorID, m.factory(), m.serializer))
+		val, _ = m.activeActors.Load(actorID)
+	}
+	return val.(ActorContainer)
+}
+
+// InvokeMethod to invoke local function by @actorID, @methodName and @request request param
+func (m *DefaultActorManager) InvokeMethod(actorID, methodName string, request []byte) ([]byte, actorErr.ActorErr) {
+	if m.factory == nil {
+		return nil, actorErr.ErrActorFactoryNotSet
+	}
+	var replyv reflect.Value
+	returnValue, aerr := m.getAndCreateActorContainerIfNotExist(actorID).Invoke(methodName, request)
```

Comment:
`returnValue, err := m.getAndCreateActorContainerIfNotExist(actorID).Invoke(methodName, request)`

Reply:
aerr is not  error type, it is ActorErr. 
If change to `err`, it would confict with err below.